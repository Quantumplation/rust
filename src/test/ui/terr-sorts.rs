struct Foo {
    a: isize,
    b: isize,
}

type Bar = Box<Foo>;

fn want_foo(f: Foo) {}
fn have_bar(b: Bar) {
    want_foo(b); //~  ERROR arguments to this function are incorrect
                 //~| expected struct `Foo`
                 //~| found struct `Box<Foo>`
}

fn main() {}
