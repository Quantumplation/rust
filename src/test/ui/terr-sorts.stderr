error[E0308]: arguments to this function are incorrect
  --> $DIR/terr-sorts.rs:10:5
   |
LL |     want_foo(b);
   |     ^^^^^^^^^-^
   |              |
   |              expected `Foo`, found `Box<Foo>`
   |
note: function defined here
  --> $DIR/terr-sorts.rs:8:4
   |
LL | fn want_foo(f: Foo) {}
   |    ^^^^^^^^ ------
help: provide an argument of the correct type
   |
LL |     want_foo({Foo});
   |     ^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
